### PR TITLE
Add catch and additional logging. #1086

### LINKED ISSUE #1086: UnhandledPromiseRejectionWarning: TypeError: Converting circular structure to JSON
## *Who* is the bug affecting?
Registry hangs on errors

## *What* is affected by this bug?
browser client, and registry server

## *When* does this occur?
During parsing error.

## *Where* on the platform does it happen?
Registry, unable to reproduce

## *How* do we replicate the issue?
Unable to reproduce

## Expected behavior (i.e. solution)
Error with logging 

## What version of OC, Node.js and OS are you using?
Registry 0.48.1, alpine linux

## Other Comments
Logs from the registry.  I am sure there is invalid code causing this, but oc should be able to handle this and not throw this error.

```
(node:1) UnhandledPromiseRejectionWarning: TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at stringify (/app/node_modules/express/lib/response.js:1119:12)
    at ServerResponse.json (/app/node_modules/express/lib/response.js:260:14)
    at async.map (/app/node_modules/oc/src/registry/routes/components.js:72:25)
```

### PR DESCRIPTION
**Pull request by kmcrawford**
Attempt to catch error and log.  Any other method of logging would be preferred.

closes #1086

### CODE REVIEW

File: src/registry/routes/components.js [hunk 1]
```diff
@@ -68,8 +68,13 @@ module.exports = function(conf, repository) {
...
       },
       (err, results) => {
-        setHeaders(results, res);
-        res.status(200).json(results);
+        try {
+          setHeaders(results, res);
+          res.status(200).json(results);
+        } catch (e) {
+          console.log(results);
+          res.status(500).json({});
```
Comment (matteofigus) [inline src/registry/routes/components.js]: Can we return the error in the form of `returnError` for consistency?
```
res.json({
  code, // Sort of indexable thing like render_error
  error // A more descriptive thing perhaps with the component name and the version?
})

Comment (kmcrawford) [reply src/registry/routes/components.js]: I could do the following to give better insight server-side & client side: 

```try {
          setHeaders(results, res);
          res.status(200).json(results);
        } catch (e) {
          console.log('RENDER_ERROR', format(
            strings.errors.registry.RENDER_ERROR,
            results.name,
            results.requestVersion),
            results.code, results.error);

          if (results.code && results.error) {
            res.status(500).json({"code": results.code, "error": results.error });
          } else {
            res.status(500).json({"code": "RENDER_ERROR", "error": format(
              strings.errors.registry.RENDER_ERROR,
                results.name,
                results.requestVersion
              )});
          }
        }
```

The is the current results payload that we are seeing in our logs when this exception is caught:
```
{ 
       code: 'GENERIC_ERROR',
       error: 'Component execution error: Request failed with status code 404',
       details: [Object],
       name: 'poorly-written-component',  //name changed to protect the innocent 
       requestVersion: '1.3.42' }
```

Comment (matteofigus) [reply src/registry/routes/components.js]: I think that would definitely improve, together with a test

**Comment by matteofigus** [comment]
@kmcrawford I made a PR to this branch to integrate the tests. Have a look 👍

**Pull request merged**